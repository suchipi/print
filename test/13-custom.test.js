const { print } = require("./helpers");

describe("print.custom", () => {
	it("can be used to modify print output", () => {
		const value = {
			something: {
				is: "my custom thing",
				[print.custom]: (inputs) => {
					inputs.type = "My custom thing!!!";
					inputs.linesAfter.push("yeah :)");
				},
			},
		};

		expect(print(value)).toMatchSnapshot();
	});

	it("receives a lot of stuff", () => {
		let inputs;

		const value = {
			something: {
				is: "my custom thing",
				[print.custom]: (inputs_) => {
					inputs = inputs_;
				},
			},
		};

		print(value, { someOpt: true });
		expect(inputs).toMatchSnapshot();
	});
});
