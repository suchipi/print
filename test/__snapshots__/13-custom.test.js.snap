// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print.custom can be used to modify print output 1`] = `
{
	something: My custom thing!!! {
		is: "my custom thing"
		Symbol(print.custom): Function "[print.custom]" {
			│1│ inputs => {
			│2│           inputs.type = "My custom thing!!!";
			│3│           inputs.linesAfter.push("yeah :)");
			│4│         }
		}
		
		yeah :)
	}
}
`;

exports[`print.custom receives a lot of stuff 1`] = `
Object {
  brackets: Array [
    {,
    },
  ],
  colours: Object {
    braces: ,
    darkGreen: ,
    date: ,
    dot: .,
    empty: empty × ,
    escape: ,
    frozen: ,
    green: ,
    grey: ,
    hexBorder: ,
    hexOffset: ,
    hexValue: ,
    keyEscape: ,
    keys: ,
    noExts: ,
    nul: ,
    nulProt: ,
    off: ,
    primitive: ,
    punct: ,
    quotes: ,
    red: ,
    reference: ,
    regex: ,
    sealed: ,
    srcBorder: ,
    srcRowNum: ,
    srcRowText: ,
    string: ,
    symbol: ,
    symbolFade: ,
    typeColour: ,
    undef: ,
  },
  indent: 	,
  key: something: ,
  linesAfter: Array [],
  linesBefore: Array [],
  oneLine: false,
  opts: Object {
    someOpt: true,
  },
  propLines: Array [
    is: "my custom thing",
    Symbol(print.custom): Function "[print.custom]" {
	│1│ inputs_ => {
	│2│           inputs = inputs_;
	│3│         }
},
  ],
  tooDeep: false,
  type: ,
  typeSuffix: ,
}
`;
